### ISSUE TITLE
Confirm email address for users

### ISSUE THREAD

**Issue by tdg** [swapmyvote/swapmyvote#15]
Confirm email address for users

**Comment by aspiers** [comment]
@tdg On further investigation I think we misunderstood this issue and that it's about confirming the user's email address.  Having said that, currently it only comes from Facebook or Twitter, so at least in theory it should have already been confirmed by one of those two platforms and we shouldn't need to confirm it.

Also, it turns out that in the production environment we use Mailgun rather than sending email directly, so deliverability based on the sender should be fine.

I think we should deal with CTR in a separate issue.

**Comment by aspiers** [comment]
Hrm, I now see it's possible for the user to update their email address.  So we need to confirm the new email address if this is done.

**Comment by aspiers** [comment]
I just logged in via Twitter and it didn't get my email.  Weird.  @jpallen Any ideas?

**Comment by aspiers** [comment]
Updated description again based on better understanding.

**Comment by tdg** [comment]
>Having said that, currently it only comes from Facebook or Twitter, so at least in theory it should have already been confirmed by one of those two platforms and we shouldn't need to confirm it.

The concern is that, because our email comes from a more obscure domain, it could easily all go to junk mail folders. Would a confirm link help voters train their email clients to treat our emails seriously and deliver to inboxes accordingly?

**Comment by aspiers** [comment]
`.uk` isn't really an obscure domain; I suspect that's not an issue.  No, email clients aren't retrained when you click a confirmation link, but at least it tells us the email address is correct.

**Comment by tdg** [comment]
Fair - but one of the last elections we did have some (ill-defined) problems with this (naked `.uk` was younger back in 2015 :) ). 

Hopefully #40 will resolve the problems, but I still feel it's a good idea to reinforce users' expectations that they will receive - and need to respond to - emails from us

**Comment by tdg** [comment]
Suggest this be moved to M0

**Comment by baob** [comment]
How does this relate to #38 ?  Does that have to be finished first ? or will #38 take care of this completely perhaps ?  @aspiers @jpallen

**Comment by aspiers** [comment]
No they are independent of each other, but email logins increase the need for verification.

**Comment by baob** [comment]
Note: not relevant to BE2022 sol long as both the twitter and facebook logins are disabled. If we manage to do #442 (twitter), this becomes important again.

**Comment by aspiers** [comment]
@baob I don't think I agree. Now we are entirely dependent on email login, but have zero validation of correct email addresses. In contrast, when we had Twitter/Facebook login, at least we could depend on those two sites validating the email addresses returned to us from their API.  However we already have this in M8 which seems correct.

**Comment by baob** [comment]
> Now we are entirely dependent on email login, but have zero validation of correct email addresses

So this also relates to #636, the whole post-registration flow.

**Comment by aspiers** [comment]
Updated for GE2024.
